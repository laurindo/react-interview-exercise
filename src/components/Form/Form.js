import React, { Component } from 'react';

export default class Form extends Component {
  render() {
    return (
      <form onSubmit={e => { e.preventDefault(); }}>
        {this.props.children}
      </form>
    );
  }
}
