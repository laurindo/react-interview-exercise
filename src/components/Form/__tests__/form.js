import React from 'react';
import { render, fireEvent, cleanup, waitForElement } from 'react-testing-library';
import Form from '../Form';

test('#02', () => {
  const {
    getByLabelText,
    getByText,
    getByTestId,
    container
  } = render(
    <Form>
      <select>
        <option>select a gender</option>
      </select>
    </Form>
  );

  expect(container.select).toMatchSnapshot();

  //expect(getByText('select a gender')).toHaveTextContent('select a gender');
});
