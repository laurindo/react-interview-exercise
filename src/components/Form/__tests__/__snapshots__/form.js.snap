exports[`test #02 1`] = `undefined`;
